spec/NewInternalsSpecification.html

```diff
@@ -189,11 +191,15 @@
       1. Set _C_'s [[PendingChangeRecords]] internal slot to a new empty List.
       1. Let _array_ be ArrayCreate(0).
       1. Let _n_ be 0.
+      1. Let _anyFound_ be *false*.
       1. For each _record_ in _changeRecords_, do:
+        1. Let _anyFound_ be *true*.
+        1. If _record_ is *undefined*, continue.
         1. Let _status_ be the result of CreateDataProperty(_array_, ToString(_n_), _record_).
         1. Assert: _status_ is *true*.
         1. Increment _n_ by 1.
-      1. If _array_ is empty, return *false*.
+      1. If _anyFound_ is *false*, return *false*.
+      1. <ins>TODO(arv): Should we call with undefined if the array is empty?</ins>
```

Comment:
Changed. Now it calls the function with either an array or undefined (the number of params is always 1).